lib/AFP/Data/ByteString/Short/Word16.hs

```diff
@@ -0,0 +1,509 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE MagicHash #-}
+{-# LANGUAGE MultiWayIf #-}
+{-# LANGUAGE PackageImports #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TupleSections #-}
+{-# LANGUAGE UnboxedTuples #-}
+{-# LANGUAGE ViewPatterns #-}
+
+module AFP.Data.ByteString.Short.Word16 (
```

Comment:
This is similar to `AFP.Data.ByteString.Short`. We use this for windows later, where we have UTF16LE encoded bytestrings and need to do word splitting.

I'm not entirely sure if this is upstreamable to `bytestring`. It seems much more specific in use case and is harder to use, since most functions expect a bytestring with an even number of bytes. See below.